### PR TITLE
Adding moth.monster (my) keyboards

### PR DESCRIPTION
**Pull request by mothdotmonster**
Hi! I've designed a few keyboards that run QMK, so I'd like to upstream them here.

## Description

This PR does adds a new keyboard subfolder containing the firmwares for the two keyboards I've designed, OK35 and VEGAS.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

***
(I apologize if I've made any mistakes in how one should make the perfect PR, I'm not an expert at git.)

### CODE REVIEW

File: keyboards/mothdotmonster/ok35/config.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#pragma once
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/mothdotmonster/ok35/config.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x86FE
+#define PRODUCT_ID      0x0035
+#define DEVICE_VER      0x0000
+#define MANUFACTURER    mothdotmonster
+#define PRODUCT         OK35
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/config.h:10]: This should be/needs tobe moved to the info/json file.

File: keyboards/mothdotmonster/ok35/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/mothdotmonster/ok35/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+  [4] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/keymaps/default/keymap.c:43]:
```suggestion
```

File: keyboards/mothdotmonster/ok35/keymaps/via/config.h [hunk 5]
```diff
@@ -0,0 +1 @@
+#define DYNAMIC_KEYMAP_LAYER_COUNT 5
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/keymaps/via/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

And needs an include guard (`#pragma once` before the contents)

File: keyboards/mothdotmonster/ok35/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/mothdotmonster/ok35/ok35.c [hunk 7]
```diff
@@ -0,0 +1 @@
+#include "ok35.h"
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/ok35.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/mothdotmonster/ok35/ok35.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
+#pragma once
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/ok35.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/mothdotmonster/ok35/ok35.h [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "quantum.h"
+
+#ifdef USE_I2C
+#include <stddef.h>
+#ifdef __AVR__
+	#include <avr/io.h>
+	#include <avr/interrupt.h>
+#endif
+#endif
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/ok35.h:11]:
```suggestion
```

File: keyboards/mothdotmonster/ok35/ok35.h [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/ok35.h:13]:
```suggestion
#define LAYOUT_ortho_5x14( \
```

File: keyboards/mothdotmonster/ok35/ok35.h [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+		{ L30, L31, L32, L33, L34, L35, L36 }, \
+		{ L40, L41, L42, L43, L44, L45, L46 }, \
+		{ R00, R01, R02, R03, R04, R05, R06 }, \
+		{ R10, R11, R12, R13, R14, R15, R16 }, \
+		{ R20, R21, R22, R23, R24, R25, R26 }, \
+		{ R30, R31, R32, R33, R34, R35, R36 }, \
+		{ R40, R41, R42, R43, R44, R45, R46 } \
+	}
+
+#define LAYOUT_ortho_5x14 LAYOUT
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/ok35.h:33]:
```suggestion
```

File: keyboards/mothdotmonster/ok35/info.json [hunk 12]
```diff
@@ -0,0 +1,85 @@
+{
+    "keyboard_name": "OK35",
+    "url": "https://moth.monster",
+    "maintainer": "mothdotmonster",
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/info.json:6]:
```suggestion
        "LAYOUT_ortho_5x14": {
```

File: keyboards/mothdotmonster/ok35/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT(
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/keymaps/default/keymap.c:5]: These should all be `LAYOUT_ortho_5x14` ideally.

File: keyboards/mothdotmonster/ok35/readme.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+# OK35
+
+![OK35](https://i.redd.it/o9gc51mxlol81.jpg)
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/readme.md:3]:
```suggestion
![OK35](https://i.redd.it/o9gc51mxlol81h.jpg)
```

File: keyboards/mothdotmonster/ok35/readme.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+
+A 5x14 split ortholinear keyboard.
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/readme.md:5]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/mothdotmonster/ok35/rules.mk [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+
+SPLIT_KEYBOARD = yes
+
+DEFAULT_FOLDER = mothdotmonster/ok35
```
Comment (drashna) [inline keyboards/mothdotmonster/ok35/rules.mk:22]: isn't needed. 
```suggestion
```

File: keyboards/mothdotmonster/vegas/readme.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
+# VEGAS
+
+![keyboard](https://i.imgur.com/VFN43xa.jpg)
```
Comment (drashna) [inline keyboards/mothdotmonster/vegas/readme.md:3]:
```suggestion
![keyboard](https://i.imgur.com/VFN43xah.jpg)
```

File: keyboards/mothdotmonster/vegas/readme.md [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+Point of sale inspired keyboard with a unique layout.
+
+* Keyboard Maintainer: mothdotmonster
+* Hardware Supported: Pro Micro and compatibles
+* Hardware Availability: [shop.moth.monster](http://shop.moth.monster/product/vegas)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mothdotmonster/vegas/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mothdotmonster/ok35/config.h [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+#define MATRIX_COL_PINS { F6, F7, B1, B3, B2, B6, F5 }
+#define SPLIT_HAND_PIN F4
+#define SOFT_SERIAL_PIN D0
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (waffle87) [inline keyboards/mothdotmonster/ok35/config.h:21]:
```suggestion
#define MATRIX_ROW_PINS { C6, D7, E6, B4, B5 }
#define MATRIX_COL_PINS { F6, F7, B1, B3, B2, B6, F5 }
#define SPLIT_HAND_PIN F4
```

File: keyboards/mothdotmonster/ok35/info.json [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+    "manufacturer": "moth.monster",
+    "url": "https://moth.monster",
+    "maintainer": "mothdotmonster",
+    "usb": {
+        "vid": "0x86FE",
+        "pid": "0x0035",
+        "device_version": "0.1.0"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/mothdotmonster/ok35/info.json:12]:
```suggestion
    "bootloader": "atmel-dfu",
    "diode_direction": "COL2ROW",
    "split" {
        "enabled": true,
        "soft_serial_pin": "D0"
    },
    "build": {
        "lto": true
    },
    "community_layouts": ["ortho_5x14"],
```

File: keyboards/mothdotmonster/ok35/rules.mk [hunk 21]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/mothdotmonster/ok35/rules.mk:6]:
```suggestion
```

File: keyboards/mothdotmonster/ok35/rules.mk [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = no      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+
+SPLIT_KEYBOARD = yes
+
```
Comment (waffle87) [inline keyboards/mothdotmonster/ok35/rules.mk:21]:
```suggestion
```

File: keyboards/mothdotmonster/ok35/rules.mk [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+DEFAULT_FOLDER = mothdotmonster/ok35
+LAYOUTS = ortho_5x14
+
+LTO_ENABLE = yes
```
Comment (waffle87) [inline keyboards/mothdotmonster/ok35/rules.mk:25]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**